src/WebJobs.Extensions.DurableTask/MaxConnectionHelper.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal static class MaxConnectionHelper
+    {
+        private const string AzureWebsiteSku = "WEBSITE_SKU";
+        private const string DynamicSku = "Dynamic";
+
+        private static bool isSet = false;
+
+        // Temporary helper until https://github.com/Azure/azure-storage-net/issues/580 is fixed.
+        public static void SetMaxConnectionsPerServer(ILogger logger)
+        {
+            if (IsDynamicSku() && !isSet)
+            {
+                Assembly storageAssembly = typeof(CloudStorageAccount).Assembly;
+                Type httpHandlerType = storageAssembly.GetType("Microsoft.WindowsAzure.Storage.Auth.Protocol.StorageAuthenticationHttpHandler");
+                PropertyInfo instanceProp = httpHandlerType?.GetProperty("Instance", BindingFlags.Static | BindingFlags.Public);
+
+                if (instanceProp?.GetValue(null) is HttpClientHandler handler)
+                {
+                    // This is a best-effort attempt. It's possible that the StorageClient
+                    // is already in-use and this will throw.
+                    try
+                    {
+                        handler.MaxConnectionsPerServer = 50;
+                    }
+                    catch (Exception ex)
+                    {
+                        logger.LogDebug($"Azure Storage HttpClientHandler was not updated: '{ex}'. Current value: {handler.MaxConnectionsPerServer}");
+                    }
+                }
+                else
+                {
+                    logger.LogDebug($"Azure Storage HttpClientHandler was not found in assembly: '{storageAssembly.FullName}'.");
```

Comment:
Can we use `LogWarning` instead of `LogDebug`? I would like it if we could see this warning more clearly if the best-effort workaround isn't working.